lib/Doctrine/Access.php

```diff
@@ -145,22 +133,19 @@ public function offsetSet($offset, $value)
     }
 
     /**
-     * Unset a given offset
-     *
-     * @see   set, offsetSet, __set
-     * @param mixed $offset
+     * @return void
      */
     #[\ReturnTypeWillChange]
     public function offsetUnset($offset)
     {
-        return $this->remove($offset);
+        $this->remove($offset);
```

Comment:
The return value is changed here. Maybe not an issue if using `unset($...)` property, but what about `if ($...->offsetUnset(...))`?

Reply:
the main interface of the ArrayAccess has the following signature: `public function offsetUnset(mixed $offset): void`, thus marking this as `@return void` and returning something is not expected

The same is already happening on the method
```    /**
     * Remove key from data
     *
     * @param   string $name
     * @return  void
     */
    #[\ReturnTypeWillChange]
    public function __unset($name)
    {
        return $this->remove($name);
    }```

and the class `Record extends Doctrine_Access` overrides that method and does not return any value.

I agree it could be a BC break, but the class is already breaking the contract with the ArrayAccess interface
What about merging as is? it it safe enough to try?

Reply:
ArrayAccess compatibility is more important than the BC break. We should make a note somewhere about this BC break.